pkg/sql/catalog/descriptor.go

```diff
@@ -763,6 +763,8 @@ type TableDescriptor interface {
 	// GetInProgressImportStartTime returns the start wall time of the in progress import,
 	// if it exists.
 	GetInProgressImportStartTime() int64
+	// Hydrated returns true iff all visible column types are hydrated.
+	Hydrated() bool
```

Comment:
Can you redefine this as a function that takes a `catalog.TableDescriptor` rather than a method? Its logic does not rely on any state that's private to the interface implementation.

Reply:
good point. will do.